airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseDestinationV1V2Migrator.kt

```diff
@@ -139,8 +137,8 @@ abstract class BaseDestinationV1V2Migrator<DialectTableDefinition> : Destination
      * @param destinationSyncMode destination sync mode
      * @return whether this is full refresh overwrite
      */
-    private fun isMigrationRequiredForSyncMode(destinationSyncMode: DestinationSyncMode?): Boolean {
-        return DestinationSyncMode.OVERWRITE != destinationSyncMode
+    private fun isMigrationRequiredForSyncMode(streamConfig: StreamConfig): Boolean {
+        return streamConfig.minimumGenerationId == 0L
```

Comment:
does it make sense to track a TruncateMode on streamconfig instead of minimumGenerationId? (totally fair if this is too much effort)

```kotlin
interface TruncateMode
data class RetainAll: TruncateMode
data class RetainLatest: TruncateMode
data class RetainSome(minimumGeneration: Long): TruncateMode
```

then all these comparisons become more clear:
* `minGeneration == 0` -> `truncateMode is RetainAll`
* `minGeneration == generation` -> `truncateMode is RetainLatest`
* `else` -> `truncateMode is RetainSome`

Reply:
maybe for another rewrite, yeah.